src/__tests__/utils/transform.tsx

```diff
@@ -1,51 +1,120 @@
+    isDirectory?: (directory: string) => boolean;
+  }
+}
+
 type ProgramTransformer = (program: ts.Program) => ts.TransformerFactory<ts.SourceFile>;
 
-/**
- * This creates a full project which will resolve all modules.
- * Only use this when wanting to test imports tbh. It's slow.
- */
-export const createFullTransform = (
-  programTransformer: ProgramTransformer,
-  dir: string
-) => (sources: {
+interface Sources {
   /**
    * This is the root file for the transform.
    */
   index: string;
   [key: string]: string;
-}): Promise<string> => {
+}
+
+/**
+ * This creates a full project which will resolve all modules.
+ * Only use this when wanting to test imports tbh. It's slow.
+ */
+export const createFullTransform = (programTransformer: ProgramTransformer) => (
+  sources: Sources
+): Promise<string> => {
   return new Promise((resolve, reject) => {
-    try {
-      fs.rmdirSync(`${dir}/.tmp`, { recursive: true });
-    } catch {}
-
-    fs.mkdirSync(`${dir}/.tmp`);
-    const files: string[] = [];
-    Object.keys(sources).forEach(key => {
-      const source = sources[key];
-      const filename = `${key}.tsx`;
-      const filepath = path.resolve(`${dir}/.tmp/${filename}`);
-      files.push(filepath);
-      fs.writeFileSync(filepath, source);
-    });
-
-    const [rootFile] = files;
+    const fs = (Volume.fromJSON(
+      Object.entries(sources).reduce<Record<string, string>>(
+        (acc, [name, content]) => ({
+          ...acc,
+          [`/${name}.tsx`]: content,
+        }),
+        {}
+      )
+    ) as unknown) as typeof Fs;
+
+    // Create mock "@untitled/css-in-js-project"
+    createMockModule(pkg.name, fs);
+
     const config: ts.CompilerOptions = {
       jsx: ts.JsxEmit.Preserve,
+      lib: ['/node_modules/typescript/lib/lib.esnext.full.d.ts'],
       module: ts.ModuleKind.ESNext,
+      moduleResolution: ts.ModuleResolutionKind.NodeJs,
       suppressImplicitAnyIndexErrors: true,
+      resolveJsonModule: true,
+      skipLibCheck: true,
       target: ts.ScriptTarget.ESNext,
+      types: [],
       // Uncomment this if shit isn't working.
       // noEmitOnError: true,
     };
+
     const compilerHost = ts.createCompilerHost(config, true);
-    const program = ts.createProgram([rootFile], config, compilerHost);
+
+    copy(
+      { fs: Fs, path: compilerHost.getDefaultLibLocation!() },
+      { fs, path: '/node_modules/typescript/lib/' }
+    );
+
+    compilerHost.getDefaultLibLocation = () => '/node_modules/typescript/lib/';
+
+    compilerHost.fileExists = file => fs.existsSync(file);
+
+    compilerHost.resolveModuleNames = (names, containingFile) => {
+      return names.map(name => {
+        return {
+          resolvedFileName: resolveModule.sync(name, {
+            basedir: path.dirname(containingFile),
+            extensions: ['.js', '.json', '.node', '.tsx', '.ts', '.d.ts'],
+            readFileSync: path => fs.readFileSync(path),
```

Comment:
so using the memfs we're like half using it? since some parts of ts still need the real fs? 🤔